File: Python/Problem_047.py
def solve(limit: int = 1_000_000) -> int:
    cnt = [0] * (limit + 1)

    for i in range(2, limit + 1):
        if cnt[i] == 0:  # prime
            for m in range(i, limit + 1, i):
                cnt[m] += 1

    streak = 0
    for n in range(2, limit + 1):
        if cnt[n] == 4:
            streak += 1
            if streak == 4:
                return n - 3
        else:
            streak = 0

    return -1

if __name__ == "__main__":
    print(solve())  # 134043

File: CPP/Problem_047.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    const int LIMIT = 1000000;
    vector<int> cnt(LIMIT + 1, 0);

    for (int i = 2; i <= LIMIT; ++i) {
        if (cnt[i] == 0) { // prime
            for (int m = i; m <= LIMIT; m += i) {
                cnt[m] += 1;
            }
        }
    }

    int streak = 0;
    for (int n = 2; n <= LIMIT; ++n) {
        if (cnt[n] == 4) {
            streak++;
            if (streak == 4) {
                cout << (n - 3) << "\n"; // 134043
                return 0;
            }
        } else {
            streak = 0;
        }
    }

    return 0;
}
